### PR TITLE
 Add IMC for registering flowers

### PR DESCRIPTION
**Pull request by temp1011**
This PR lets mods register their own flowers with forestry. This works a lot better than doing it through plugins on the forestry side since another mod has much better access to it's own blockstates.

For example, in https://github.com/ForestryMC/ForestryMC/issues/2022 rather than forestry checking when an agricraft config option is disabled then finding the correct blockstate and registering it there could be a plugin for Forestry in agricraft to just send an imc if the config option is changed (I would be happy to write this plugin).

### CODE REVIEW

File: src/main/java/forestry/apiculture/ModuleApiculture.java [hunk 1]
```diff
@@ -988,6 +990,39 @@ public boolean processIMCMessage(IMCMessage message) {
 				HiveConfig.addBlacklistedDim(dim);
 			}
 			return true;
+		} else if (message.key.equals("add-plantable-flower")) {
+			try {
```
Comment (mezz) [inline src/main/java/forestry/apiculture/ModuleApiculture.java]: This method is getting very long, can you separate out these new cases into new methods?

**Review by mezz — CHANGES_REQUESTED** [review]
Looks good, I like the idea.

**Comment by temp1011** [comment]
Should the `try/catch` be inside or outside the method call? Inside looks a bit weird but outside increases the other method length.

**Comment by mezz** [comment]
Inside is fine I think.

**Review by mezz — APPROVED** [review]

**Pull request merged**